test/entry_test.py

```diff
@@ -1 +1,53 @@
-import entry
\ No newline at end of file
+import logging
+import os
+import shutil
+import unittest
+from autologging import logged
+from datetime import datetime
+from exam_date.stored_date import AssignmentLatestSubmittedDate
+
+logging.basicConfig(level=os.getenv("log_level", "DEBUG"))
+
+
+@logged
+class TestSPEProcess(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.__log.info("you got into the testing zone")
+        this_dir = os.path.dirname(os.path.abspath(__file__))
+        self.persisted_dir: str = this_dir + '/' + 'PERSIST/'
+        self.file_name: str = 'test_persisted.txt'
+
+    def _check_date_format(self, utc_date: str):
+        try:
+            datetime.strptime(utc_date, '%Y-%m-%dT%H:%M:%SZ')
```

Comment:
Doesn't Python have a predefined format for UTC?

Reply:
```
>>> from datetime import datetime
>>> datetime.utcnow()
datetime.datetime(2019, 1, 2, 20, 45, 20, 625635)
```
It is returning a object type and from that we need to use the `strptime` to get the required format

Reply:
I was referring to the "`%Y-%m-%dT%H:%M:%SZ`" format you used with `strptime()`.  I thought that DateTime may provide a format for ISO 8601 format.  For example, in PHP we can use the format "`%c`" to get nearly the same thing as your format. (http://php.net/manual/en/function.date.php#refsect1-function.date-parameters)

I looked up Python's datetime formatting: https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior

It appears datetime doesn't have an ISO 8601 format defined.  However, it does have a function for doing just this, `isoformat()`: https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat.  You should use that rather than specifying the format yourself.  If it adds a time zone offset to the result and you'd like to avoid that, you'll need to set the UTC offset to zero (set `tzinfo` to `None`) before you call `isoformat()`.  I suspect your time doesn't have a TZ at that point anyway.

Reply:
Nice info! The function is expecting in the date like `date = datetime.fromisoformat('2017-01-01T12:30:59.000000')`. if i used that function for my purposes then it will throw the error. 
```
>>> datetime.fromisoformat('2019-01-03T22:52:39Z')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Invalid isoformat string: 2019-01-03T22:52:39Z
```

I am looking for a specific format with `z` as that's what canvas sends back for test taken date

Reply:
Yes, I guess that's the chance we take when using code that "supports" ISO 8601 timestamps.  I think some developers prefer to support either no timezone offset or those with numeric values.  They don't want to also support the "Z" option.

The documentation for `fromisoformat()` states:

> **Caution:** This does not support parsing arbitrary ISO 8601 strings - it is only intended as the inverse operation of `datetime.isoformat()`.

So there was the chance it wasn't going to work for your purposes, but we couldn't be sure whether it would until you'd tried it.  The documentation for `isoformat()` doesn't indicate it will output "Z", so I guess that's an indication `fromisoformat()` wouldn't parse it, either.